### PR TITLE
Upgrade java

### PR DESCRIPTION
**Pull request by this-Aditya**

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -24,7 +24,7 @@ jobs:
 
       - uses: actions/setup-java@v3
         with:
-          java-version: 11
+          java-version: 17
```
Comment (blootsvoets) [inline .github/workflows/main.yml:27]: Good find. Could you please also change the `java-version` in all the other GitHub workflows to 17?

**Review by blootsvoets — COMMENTED** [review]
Nice! Could you please rebase with dev to get a clean PR?

**Comment by this-Aditya** [comment]
Done.

**Review by blootsvoets — APPROVED** [review]
LGTM, thanks for the changes!

**Pull request merged**